data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/eventtype/EventTypeCreatorImpl.java

```diff
@@ -92,7 +92,32 @@ public Future<EventType> create(
                     .withSchema(event.getDataSchema())
                     .withDescription("Event Type auto-created by controller");
 
-            return this.eventTypeClient.resource(et.build()).create();
+            try {
+                return this.eventTypeClient.resource(et.build()).create();
+            } catch (io.fabric8.kubernetes.client.KubernetesClientException e) {
+                // Handle race condition where EventType was created between cache check and creation attempt
+                // Check for HTTP 409 (Conflict) status code or "AlreadyExists" reason
+                final boolean isAlreadyExists = (e.getCode() == 409)
+                        || (e.getStatus() != null
+                                && "AlreadyExists".equals(e.getStatus().getReason()));
```

Comment:
generelly i am big fan of using library constants for this - where possible.

Just saying

Reply:
I agree, but I didn't see anything specific in fabric8 for this (maybe I missed it).